main.cpp

```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
+            std::cout << "Pencil sharpened" << std::endl;
+        }	
+        else if(sharpness >= 2.5f)
+        {
+            sharpEnough = true;
+            std::cout << "Super sharp!" << std::endl;
+        }
+        else
+        {
+            sharpEnough = false;
+            std::cout << "Needs more sharpening" << std::endl;
+        }	
+    }
+    else
+        std::cout << "Pencil sharp enough" << std::endl;
+}
+
+void PencilCase::Pencil::writeSomething(std::string writing)
+{
+    std::cout << writing << std::endl;
+    std::cout << "Finished writing" << std::endl;
+    sharpness-= 0.089f;
+}
+
+bool PencilCase::Pencil::isSharp()
+{
+    return sharpEnough;
+}
+
+// ===============PencilCase Functions=================
+
+PencilCase::PencilCase() : maxItemsToHold(20), isOpen(false)
+{
+    std::cout << "****** PencilCase Created ******" << std::endl;	
+}
+
+PencilCase::~PencilCase()
+{
+    std::cout << "****** PencilCase Destroyed ******" << std::endl;
+}
+
+void PencilCase::open()
+{
+    isOpen = true;
+}
+
+void PencilCase::close()
+{
+    isOpen = false;
+}
+
+void PencilCase::addItems(int itemsToAdd)
+{
+    if(isOpen)
+    {
+        std::cout << "Adding items to pencilcase..." << std::endl;
+
+        while(numItemsAdded < maxItemsToHold && itemsToAdd > 0)
+        {
+            ++numItemsAdded;
+            --itemsToAdd;
+        }
+        (numItemsAdded == maxItemsToHold) ?  std::cout << "Pencilcase full!" << std::endl : std::cout << numItemsAdded << " items added..." << std::endl;
+    }
+    else
+        std::cout << "Open the pencilcase first" << std::endl;
+}
+
+void PencilCase::removeItems(int itemsToRemove)
+{
+    if(itemsToRemove > 0)
+    {
+        if(isOpen && numItemsAdded > 0)
+        {
+            std::cout << "Items leaving the pencilcase..." << std::endl;
+
+            while(numItemsAdded > 0 && itemsToRemove > 0)
+            {
+                numItemsAdded--;
+                itemsToRemove--;
+            }
+            std::cout << numItemsAdded << " items remaining" << std::endl;
+        }
+        else
+            std::cout << "Pencilcase is closed or nothing inside" << std::endl;
+    }
+    else
+        std::cout << "Can't remove " << itemsToRemove << " items"  << std::endl;
+}
+
+// ============= Drill Functions =================
+
+Drill::Drill() : isPlugged(true), drillbitAttached(false), powerInWatts(3600) 
+{
+    std::cout << "****** Drill Created ******" << std::endl;
+}
+
+Drill::~Drill()
+{
+    std::cout << "****** Drill Destroyed ******" << std::endl;
+}
+
+void Drill::drillHole(int holes)
+{
+    if(isPlugged && drillbitAttached && holes > 0)
+    {
+        std::cout << "Drilling holes..." << std::endl;
+        std::cout << "Drilling done!!" << std::endl;
+    }
+    else if(holes < 1)
+        std::cout << "You can't drill " << holes << " holes..." << std::endl;
+    else 
+        std::cout << "Whoops! Check if your drill is ready" << std::endl;
```

Comment:
If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [ ] revise